frontstage/controllers/case_controller.py

```diff
@@ -261,3 +210,61 @@ def validate_case_category(category):
         raise InvalidCaseCategory(category)
 
     logger.debug('Successfully validated case category', category=category)
+
+
+def get_cases_for_list_type_by_party_id(party_id, list_type='todo'):
+    logger.debug('Get cases for party for list', party_id=party_id, list_type=list_type)
+
+    business_cases = get_cases_by_party_id(party_id)
+    cases = filter_cases_by_case_group(business_cases)
+    history_statuses = ['COMPLETE', 'COMPLETEDBYPHONE', 'NOLONGERREQUIRED']
+    if list_type == 'history':
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case.get('caseGroup', {}).get('caseGroupStatus') in history_statuses]
+    else:
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case.get('caseGroup', {}).get('caseGroupStatus') not in history_statuses]
+
+    logger.debug("Sucessfully retrieved cases for party survey list", party_id=party_id, list_type=list_type)
+    return filtered_cases
+
+
+def filter_cases_by_case_group(cases):
+    for key, group in itertools.groupby(cases, key=lambda x: x['caseGroup']['id']):
+        group = list(group)
+        if len(group) > 1:
+            sorted_group = sorted(group, key=lambda k: k['createdDateTime'], reverse=True)
+            for i in sorted_group:
+                if sorted_group.index(i) != 0:
+                    cases.remove(i)
+    return cases
+
+
+def link_enrolment_to_cases(enrolment, cases):
+    case = next((case for case in cases
+                if enrolment['business_party']['id'] == case['partyId']
+                and enrolment['collection_exercise']['id'] == case['caseGroup']['collectionExerciseId']), None)
+    if case:
+        collection_instrument = collection_instrument_controller.get_collection_instrument(case['collectionInstrumentId'])
+        return {**enrolment,
+                "case_id": case['id'],
+                "collection_instrument": collection_instrument,
+                "status": calculate_case_status(case['caseGroup']['caseGroupStatus'], collection_instrument['type'])
+                }
+    else:
+        return
+
+
+def get_enrolments_with_cases(enrolments, tag):
+    business_ids = {enrolment['business_party']['id']
+                    for enrolment in enrolments}
+    cases = [get_cases_for_list_type_by_party_id(business_id, tag)
+             for business_id in business_ids]
+    flattened_case_list = [case for sublist in cases for case in sublist]
```

Comment:
We do this flattened thing in a few places. We could make it a utility function maybe?
Something like:
```
def flatten_list(list):
    return [item for sublist in list for item in sublist]
```